src/panels/config/devices/ha-config-device-page.ts

```diff
@@ -1119,17 +1119,32 @@ export class HaConfigDevicePage extends LitElement {
       const matter = await import(
         "./device-detail/integration-elements/matter/device-actions"
       );
-      const actions = await matter.getMatterDeviceActions(
+      const defaultActions = matter.getMatterDeviceDefaultActions(
         this,
         this.hass,
         device
       );
-      deviceActions.push(...actions);
+      deviceActions.push(...defaultActions);
+      this._loadMatterDeviceActions(matter, device);
     }
 
     this._deviceActions = deviceActions;
   }
 
+  // load matter device actions async to avoid an UI with 0 actions when the matter integration needs very long to get node diagnostics
+  private async _loadMatterDeviceActions(
+    matter: typeof import("./device-detail/integration-elements/matter/device-actions"),
+    device: DeviceRegistryEntry
+  ): Promise<void> {
+    const actions = await matter.getMatterDeviceActions(
+      this,
+      this.hass,
+      device
+    );
+
+    this._deviceActions = [...actions, ...(this._deviceActions || [])];
```

Comment:
Does this order matter? Because you now add them to the front of the array where previously they where at the back.

Reply:
Yeah I add it at the front that it behaves like before. 
When I would keep default actions in the front, `ping device` would be the default action instead of `share device` when the diagnostics are loaded as normal.